larcorealg/Geometry/GeometryCore.cxx

```diff
@@ -1353,6 +1353,16 @@ namespace geo {
 
   } // GeometryCore::IntersectSegments()
 
+
+  //......................................................................
+  geo::Point_t GeometryCore::WiresIntersect
+    (geo::WireGeo const& wire1, geo::WireGeo const& wire2) const
+  {
+    assert(!wire2.isParallelTo(wire1));
+    return geo::WiresIntersection(wire1, wire2);
+  } // GeometryCore::WiresIntersect()
```

Comment:
As this function does not rely on `GeometryCore`'s state, it should not be a member function--yes, this breaks with "prior art", but I have a comment there, too.  The `geo::WiresIntersection` free function should be sufficient.  

Additionally, this member function appears to be a wrapping of the free function--however, there is a precondition that is checked here (i.e. the assert) that is already checked in the underlying implementation (`LineClosestPointWithUnitVectors`).

Reply:
> this function does not rely on GeometryCore's state

Practically, this is not true: all `geo::WireGeo` instances in LArSoft are owned by `geo::GeometryCore`. Does this change your statement?

Reply:
No, my statement is unchanged.  There is nothing in the `geo::WiresIntersection(a, b)` function that requires a `geo::GeometryCore` object--everything is passed in as function arguments and no `this` pointer is required.  The intersection function does not care where those `geo::WireGeo` objects came from nor who owns them.  And that's good--it's easier to test things and demonstrates a better separation of concerns.